binderhub/repoproviders.py

```diff
@@ -64,6 +66,36 @@ def get_build_slug(self):
         return '{user}-{repo}'.format(user='Rick', repo='Morty')
 
 
+class GitRepoProvider(RepoProvider):
+    """Bare bones git repo provider. Users must provide a resolve_ref in the following form
+
+    http://hostname.com/path/repo_path#sha1
+    """
+
+    name = Unicode("git")
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        parsed = urllib.parse.urlparse(self.spec)
+        if not parsed.fragment:
+            raise ValueError("spec used for raw git api must include the sha1 as a fragment")
+        self.resolved_ref = parsed.fragment
```

Comment:
There should probably be some validation here, unless the errors out of a failed build to a bad ref are good enough. Plus, the fragment may not be reliable because the server doesn't tend to get that part of the URL, so it would have to be escaped somehow.

Reply:
yeah, i'm planning to add some basic checks to at least enforce that it is a sha1